.github/workflows/rpm-build-and-test.yml

```diff
@@ -411,70 +224,95 @@ jobs:
+            fi
+            case '${{ matrix.distro }}' in
                 'el8')
                     CHROOT_NAME="rocky+epel-8-x86_64"
                     DISTRO_NAME="EL"
                     DISTRO_NAME_UPPER="EL"
                     DISTRO_NAME_LOWER="el"
-                    DISTRO_VERSION="${{ env.CP_EL8_VERSION &&
-                                        env.CP_EL8_VERSION || env.EL8_VERSION }}"
+                    DISTRO_VERSION='${{ env.CP_EL8_VERSION &&
+                                        env.CP_EL8_VERSION || env.EL8_VERSION }}'
                     DISTRO_VERSION_MAJOR="8"
                     OPENMPI="openmpi"
-                    LABEL="${{ env.CP_EL8_VM9_LABEL &&
-                               env.CP_EL8_VM9_LABEL || 'ci_vm9' }}"
+                    LABEL='${{ env.CP_EL8_VM9_LABEL &&
+                               env.CP_EL8_VM9_LABEL || 'all_vm9' }}'
                     ;;
                 'el9')
                     CHROOT_NAME="rocky+epel-9-x86_64"
                     DISTRO_NAME="EL"
                     DISTRO_NAME_UPPER="EL"
                     DISTRO_NAME_LOWER="el"
-                    DISTRO_VERSION="${{ env.CP_EL9_VERSION &&
-                                        env.CP_EL9_VERSION || env.EL9_VERSION }}"
+                    DISTRO_VERSION='${{ env.CP_EL9_VERSION &&
+                                        env.CP_EL9_VERSION || env.EL9_VERSION }}'
                     DISTRO_VERSION_MAJOR="9"
                     PROV_DISTRO_VERSION_MAJOR="8"
                     OPENMPI="openmpi"
-                    LABEL="${{ env.CP_EL9_VM9_LABEL &&
-                               env.CP_EL9_VM9_LABEL || 'ci_vm9' }}"
+                    LABEL='${{ env.CP_EL9_VM9_LABEL &&
+                               env.CP_EL9_VM9_LABEL || 'all_vm9' }}'
                     ;;
                 'leap15')
-                    CHROOT_NAME="opensuse-leap-${{ env.CP_LEAP15_VERSION &&
+                    CHROOT_NAME='opensuse-leap-${{ env.CP_LEAP15_VERSION &&
                                                    env.CP_LEAP15_VERSION ||
-                                                   env.LEAP15_VERSION }}-x86_64"
+                                                   env.LEAP15_VERSION }}-x86_64'
                     DISTRO_NAME="Leap"
                     DISTRO_NAME_UPPER="LEAP"
                     DISTRO_NAME_LOWER="leap"
-                    DISTRO_VERSION="${{ env.CP_LEAP15_VERSION &&
-                                        env.CP_LEAP15_VERSION || env.LEAP15_VERSION }}"
+                    DISTRO_VERSION='${{ env.CP_LEAP15_VERSION &&
+                                        env.CP_LEAP15_VERSION || env.LEAP15_VERSION }}'
                     DISTRO_VERSION_MAJOR="15"
                     OPENMPI="openmpi3"
-                    LABEL="${{ env.CP_LEAP15_VM9_LABEL &&
-                               env.CP_LEAP15_VM9_LABEL || 'ci_vm9' }}"
+                    LABEL='${{ env.CP_LEAP15_VM9_LABEL &&
```

Comment:
I'm a bit confused by the change from double quotes to single quotes when there's a single quoted string inside. Does that not cause issues?

Reply:
In this case, because the single quoted string on the inside is in a GHA _expression_ they are consumed by GHA runner and it's the resulting value of `env.CP_LEAP15_VM9_LABEL` or `all_vm9` (without the quotes) that the shell ends up seeing inside of the outer single quotes.